VimeoNetworking/Sources/VimeoClient.swift

```diff
@@ -454,15 +460,39 @@ final public class VimeoClient
     
     // MARK: - Private error handling
     
-    private func handleError<ModelType: MappableResponse>(error: NSError, request: Request<ModelType>)
+    private func handleError<ModelType: MappableResponse>(error: NSError, request: Request<ModelType>, task: NSURLSessionDataTask? = nil)
     {
         if error.isServiceUnavailableError
         {
             Notification.ClientDidReceiveServiceUnavailableError.post(object: nil)
         }
         else if error.isInvalidTokenError
         {
-            Notification.ClientDidReceiveInvalidTokenError.post(object: nil)
+            Notification.ClientDidReceiveInvalidTokenError.post(object: self.token(fromTask: task))
+        }
+    }
+    
+    private func token(fromTask task: NSURLSessionDataTask?) -> String?
+    {
+        guard let task = task else
+        {
+            return nil
+        }
+        
+        if let bearerHeader = task.originalRequest?.allHTTPHeaderFields?[Constants.AuthorizationHeader]
+        {
+            guard let range = bearerHeader.rangeOfString(Constants.BearerQuery) else
+            {
+                return nil
+            }
+            
+            var mutableBearerHeader = String(bearerHeader)
```

Comment:
same thing here, I think this can be cleaned up a bit:

return String(bearerHeader).removeRange(range)